pkg/controller/certificates/rootcacertpublisher/metrics.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rootcacertpublisher
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// RootCACertPublisher - subsystem name used by root_ca_cert_publisher
+const RootCACertPublisher = "root_ca_cert_publisher"
+
+var (
+	syncCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      RootCACertPublisher,
+			Name:           "sync_total",
+			Help:           "Number of namespace syncs happened in root ca cert publisher.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"namespace", "code"},
```

Comment:
@logicalhan label per namespace has cardinality issues, right?

Reply:
Yes it's essentially like 'resources' in the request metrics. But this tends to be problematic only in aggregate. We won't be able to promote this one to STABLE status until we have the mechanism for latching an explicit expected valueset on metric labels. @YoyinZyc has agreed to take that feature on sometime in the near future: https://github.com/kubernetes/kubernetes/issues/76302 for tracking.

Reply:
> Yes it's essentially like 'resources' in the request metrics. But this tends to be problematic only in aggregate. We won't be able to promote this one to STABLE status until we have the mechanism for latching an explicit expected valueset on metric labels.

I can't tell if you're saying this is problematic and should be dropped, or is ok for now because https://github.com/kubernetes/kubernetes/issues/76302 will do something to improve things in the future?

Reply:
It is problematic, but we have worse offenders currently, which means that if this becomes problematic, then other metrics (with considerable operational significance) will be suffering as well. I guess what I'm trying to say is that it's probably safe to wave this in, because if this is a problem, then we have other really big problems already committed to k8s. 

https://github.com/kubernetes/kubernetes/issues/76302 will improve this front in the future and provide an escape hatch.

Reply:
which is a better approach:
* adding a namespace dimension here to start and hoping #76302 gets resolved we want to promote this metric to stable?
* starting with just `code` which can be monitored/aggregated to indicate an issue, and adding `namespace` later before promoting to stable if #76302 looks to be resolved

Reply:
I'd opt for 1, for simplicity. We're introducing two escape hatches: the first (targeted in this release) allows us to shunt a metric off completely, while the second will allow us to explicitly bind values for a label to an expected set of label values (that will probably land in the following release).

Reply:
ok